main.go

```diff
@@ -111,8 +116,15 @@ func run(c *cli.Context) error {
 		go func() { startchan <- struct{}{} }()
 		file := c.String("file")
 		configchan := make(chan bool)
+		// XXX Unwatchable Puppet sources should get events from a timer,
+		// with the same interval as the Puppet agent
+		// (hint: puppet config print runinterval)
 		if !c.Bool("no-watch") {
-			configchan = ConfigWatch(file)
+			if c.IsSet("file") {
+				configchan = ConfigWatch(file)
+			} else if strings.HasSuffix(c.String("puppet"), ".pp") {
+				configchan = ConfigWatch(c.String("puppet"))
```

Comment:
OK, that also makes sense.

One thing I'm worried about. If we choose to *not* watch the manifest file, `mgmt` will ignore changes and pick them up eventually, once the 30 minute mark passed. I suspect that this behavior will surprise users. It is also inconvenient (assuming that the user does not want to restart the `mgmt` process to force a retranslation. Please note that, on the other hand, this compromise would likely be quite acceptable to most...we kick services all the time, so why not this one.)